pkg/sql/schemachanger/scbuild/internal/scbuildstmt/create_index.go

```diff
@@ -212,6 +215,131 @@ func nextRelationIndexID(b BuildCtx, relation scpb.Element) catid.IndexID {
 	}
 }
 
+func newUndefinedOpclassError(opclass tree.Name) error {
+	return pgerror.Newf(pgcode.UndefinedObject, "operator class %q does not exist", opclass)
+}
+
+func checkColumnAccessibilityForIndex(colName string, columnElts ElementResultSet, store bool) {
+	_, _, column := scpb.FindColumn(columnElts)
+	if column.IsInaccessible {
+		panic(pgerror.Newf(
+			pgcode.UndefinedColumn,
+			"column %q is inaccessible and cannot be referenced",
+			colName))
+	}
+
+	if column.IsSystemColumn {
+		if store {
+			panic(pgerror.Newf(
+				pgcode.FeatureNotSupported,
+				"index cannot store system column %s",
+				colName))
+		} else {
+			panic(pgerror.Newf(
+				pgcode.FeatureNotSupported,
+				"cannot index system column %s",
+				colName))
+		}
+	}
+}
+
```

Comment:
nit: similarly, can you some comment to explain what this function does in a high level?

Reply:
Question: Is this logic copied/mimiced from somewhere else?

Reply:
it should be :) it's mostly from `create_index.go` of legacy schema changer, for example: https://github.com/cockroachdb/cockroach/blob/b6e661e4ff03bd7c9650c703f1fb98dc213cad63/pkg/sql/create_index.go#L330-L330

Reply:
Added the comment:

// checkColumnAccessibilityForIndex validate that any columns that are explicitly referenced in a column for storage or as a key are either accessible and not system columns.

Reply:
> Question: Is this logic copied/mimiced from somewhere else?

Yes, this is taken from, checkIndexColumns